src/ORM/BehaviorRegistry.php

```diff
@@ -264,20 +265,30 @@ public function call(string $method, array $args = []): mixed
     /**
      * Invoke a finder on a behavior.
      *
+     * @internal
      * @param string $type The finder type to invoke.
-     * @param array $args The arguments you want to invoke the method with.
+     * @param \Cake\ORM\Query\SelectQuery $query The query object to apply the finder options to.
+     * @param mixed ...$args Arguments that match up to finder-specific parameters
      * @return \Cake\ORM\Query\SelectQuery The return value depends on the underlying behavior method.
      * @throws \BadMethodCallException When the method is unknown.
      */
-    public function callFinder(string $type, array $args = []): SelectQuery
+    public function callFinder(string $type, SelectQuery $query, mixed ...$args): SelectQuery
     {
         $type = strtolower($type);
 
         if ($this->hasFinder($type) && $this->has($this->_finderMap[$type][0])) {
             [$behavior, $callMethod] = $this->_finderMap[$type];
-            $callable = [$this->_loaded[$behavior], $callMethod];
+            $callable = $this->_loaded[$behavior]->$callMethod(...);
+
+            if (!$args) {
+                $reflected = new ReflectionFunction($callable);
+                $param = $reflected->getParameters()[1] ?? null;
+                if ($param?->name === 'options' && !$param->isDefaultValueAvailable()) {
+                    $args = [[]];
```

Comment:
This adds an empty options array when calling find with no arguments and the finder expects $options.

Reply:
It's quite unfortunate that the `$options` arg doesn't default to an empty array currently. It would have made no practical difference.

Perhaps we can add the default value now to core finders and throw a warning when this `if` returns true so that we can at least remove this reflection in the future.

Reply:
> Perhaps we can add the default value now to core finders and throw a warning when this if returns true so that we can at least remove this reflection in the future.

This sounds like a great inclusion into 4.5.0. We could do the detection and deprecation warning there, and in 5.x we can enforce a default options value, or we can start spreading options as named parameters.

I think the point I want to make, is that if we're going to break compatibility why not do what would be best going forward, and then we can figure out how we add deprecations and backporting to 4.5 to make the upgrade process understandable and incrementally adoptable.

Reply:
Making $options options is definitely complicated. It looks like code we don't want to support.